src/Text/Pretty/Simple/Internal/OutputPrinter.hs

```diff
@@ -121,81 +130,116 @@ hCheckTTY h options = liftIO $ conv <$> tty
     tty :: IO Bool
     tty = hIsTerminalDevice h
 
--- | Given 'OutputOptions' and a list of 'Output', turn the 'Output' into a
--- lazy 'Text'.
-render :: OutputOptions -> [Output] -> Text
-render options = toLazyText . foldr foldFunc "" . modificationsOutputList
-  where
-    foldFunc :: Output -> Builder -> Builder
-    foldFunc output accum = runReader (renderOutput output) options `mappend` accum
+-- | Parse a string, and generate an intermediate representation,
+-- suitable for passing to any /prettyprinter/ backend.
+-- Used by 'Simple.pString' etc.
+layoutString :: OutputOptions -> String -> SimpleDocStream AnsiStyle
+layoutString opts = exprsToDocStream opts . expressionParse
```

Comment:
This is potentially very useful, and should perhaps actually be exported from a non-Internal module somewhere.

It makes things like [coloured HTML output](https://github.com/cdepillabout/pretty-simple/issues/50) much easier.

Reply:
> It makes things like coloured HTML output much easier.

Well, it would, except that `AnsiStyle` is opaque, and so it wouldn't be possible to define a conversion from it.

Not sure how best to resolve this. Perhaps have our own abstract Colour type?